### PR TITLE
Task/org unit dialog

### PR DESCRIPTION
**Pull request by neeilya**
Fixes DHIS2-5207.

Changes proposed in this pull request:

- Ability to select org unit children using context menu
- Adjusted stylings
    - cursor: pointer for org unit labels & folder icons
    - Align left user org units

### CODE REVIEW

File: packages/org-unit-dialog/src/OrgUnitSelector.js [hunk 1]
```diff
@@ -70,8 +76,36 @@ class OrgUnitSelector extends Component {
...
+        if (!hasChildren) {
+            return;
+        }
+
+        this.setState({
+            menuAnchorElement: event.currentTarget,
+            loadingChildren: true,
+        });
+
+        loadChildren().then((children) => {
```
Comment (jenniferarnesen) [inline packages/org-unit-dialog/src/OrgUnitSelector.js]: Should this statement be a callback to the setStatement above?

File: packages/org-unit-dialog/src/OrgUnitSelector.js [hunk 2]
```diff
@@ -166,13 +200,27 @@ class OrgUnitSelector extends Component {
...
+                            anchorEl={this.state.menuAnchorElement}
+                            open={Boolean(this.state.menuAnchorElement)}
+                            onClose={this.closeContextMenu}
+                        >
+                            <MenuItem
+                                onClick={this.selectChildren}
+                                disabled={this.state.loadingChildren}
+                                dense
+                            >
+                                Select children
```
Comment (jenniferarnesen) [inline packages/org-unit-dialog/src/OrgUnitSelector.js]: 'Select children' should be internationalized

File: packages/org-unit-dialog/src/OrgUnitDialog.js [hunk 3]
```diff
@@ -116,6 +117,12 @@ OrgUnitDialog.propTypes = {
     */
     handleUserOrgUnitClick: PropTypes.func.isRequired,
 
+    /**
+     * Callback handler for selecting user orgunit
+     * Arguments supplied in callback: event, checked
+     */
+    handleMultipleOrgUnitsSelect: PropTypes.func.isRequired,
```
Comment (jenniferarnesen) [inline packages/org-unit-dialog/src/OrgUnitDialog.js]: So this will be a breaking change then.

**Comment by jenniferarnesen** [comment]
The code seems fine, but OrgUnitTree.component just keeps getting bigger and more complex. Can you spend some time doing some basic refactoring and adding tests? It can be done in a separate PR, but would be good to do this before picking up a new task.

Thoughts on refactoring:

- move styles to a separate file
- pull the loading function into a separate services.js file
- the render function has a bunch of comments that are unnecessary
- the render function has a high (poor) complexity score. Any way to simplify it? (e.g., make a OUCheckbox component, OUFolderIcon component), pull the event handlers out to standalone functions with argument...

File: packages/org-unit-dialog/src/OrgUnitDialog.js (hunk 3 above)
Comment (neeilya) [reply packages/org-unit-dialog/src/OrgUnitDialog.js]: I can make it optional, just wondering if it causes problems?

**Pull request merged**